apache/superset
# feat(explore): Dataset Panel Options when Source = Query

### SUMMARY

This adds a different dataset panel dropdown when a chart's datasource is a query.


### SCREENSHOTS / ANIMATED GIF

<img width="695" alt="queryDatasourcePanel" src="https://user-images.githubusercontent.com/55605634/169922171-4ff59a63-8d69-4b44-8351-44611c76e967.png">
<img width="1339" alt="queryDatasourceSaveDatasetModal" src="https://user-images.githubusercontent.com/55605634/169922178-cacf88c2-d6ac-4cfc-857a-b34465812f0e.png">

![queryDatasetPanel](https://user-images.githubusercontent.com/55605634/169922192-b754d665-d936-4967-9f08-617575e24695.gif)


### TESTING INSTRUCTIONS

- Open a chart with a query datasource type
- Click the thee dots next to the chart name to reveal the dataset options dropdown
- Observe that the options are now "Query preview", "View in SQL Lab", and "Save as dataset"
- Click "View in SQL Lab" to be taken to SQL Lab
- Click "Save as dataset" and observe the save dataset modal

### ADDITIONAL INFORMATION
